libraries/resource_providers/api500/c7000/storage_pool_provider.rb

```diff
@@ -18,20 +18,19 @@ module C7000
       class StoragePoolProvider < API300::C7000::StoragePoolProvider
         include OneviewCookbook::RefreshActions::RequestRefresh
 
+        def refresh
+          load_storage_system
+          super
+        end
+
         def load_storage_system
           raise("Unspecified property: 'storage_system'. Please set it before attempting this action.") unless @new_resource.storage_system
-          data = {
-            hostname: @new_resource.storage_system,
-            name: @new_resource.storage_system
-          }
-          @item.set_storage_system(load_resource(:StorageSystem, data))
+          @item['storageSystemUri'] = load_resource(:StorageSystem, { hostname: @new_resource.storage_system, name: @new_resource.storage_system }, :uri)
         end
 
         def update_manage_state(managed)
-          status = 'managed'
-          status = 'unmanaged' unless managed
-          return Chef::Log.info("#{@resource_name} '#{@name}' is already #{status}") if @item['isManaged'] == managed
-          @context.converge_by "#{@resource_name} '#{@name}' is now #{status}" do
+          return Chef::Log.info("#{@resource_name} '#{@name}' managed status is already #{managed}") if @item['isManaged'] == managed
+          @context.converge_by "#{@resource_name} '#{@name}' managed status is now #{managed}" do
```

Comment:
This message does not makes much sense "_Potato Potato_ **managed status is now false**"